common/src/main/java/org/jboss/hal/testsuite/dmr/ResourceVerifier.java

```diff
@@ -119,16 +139,29 @@ public void verifyAttribute(ResourceAddress address, String attribute, String ex
 
     /**
      * Verifies the attribute against the {@code expected} parameter using the specified timeout.
+     * Trying to get response with expected value until specified timeout expires
+     * RECOMMENDATION : It is not good to set timeout bigger than 5000 miliseconds
      */
     public void verifyAttribute(ResourceAddress address, String attribute, String expected, int timeout) {
-        DmrResponse response = dispatcher.execute(readAttributeOperation(address, attribute, timeout));
+        DmrResponse response = dispatcher.execute(readAttributeOperation(address, attribute));
+        long start = System.currentTimeMillis();
+
+        while (System.currentTimeMillis() <= start + timeout && !expected.equals(response.payload().asString())) {
+            response = getResponseAndWait(address, attribute);
+        }
+
         assertTrue(response.isSuccessful());
         assertEquals(expected, response.payload().asString());
     }
 
-    private Operation readAttributeOperation(ResourceAddress address, String attribute, int timeout) {
-        return new Operation.Builder(READ_ATTRIBUTE_OPERATION, address)
-                .param(ModelDescriptionConstants.NAME, attribute)
-                .withTimeout(timeout).build();
+    private Operation readAttributeOperation(ResourceAddress address, String attribute) {
+            return new Operation.Builder(READ_ATTRIBUTE_OPERATION, address)
+                    .param(ModelDescriptionConstants.NAME, attribute).build();
+    }
+
+    private DmrResponse getResponseAndWait(ResourceAddress address, String attribute) {
+        DmrResponse response = dispatcher.execute(readAttributeOperation(address, attribute));
+        Library.letsSleep(50);
```

Comment:
Please switch order of lines 163 and 164 (move `sleep` before `dispatcher.execute`) to avoid waiting for 50ms longer than is necessary.